swap_meet/vendor.py

```diff
@@ -1,2 +1,229 @@
+
+        #need to add first to both and then remove otherwise it won't work...
+        #can just use self.swap_items and pass in the first items of each list
+        self.add(friend.inventory[0])
+        friend.add(self.inventory[0])
+        
+        self.remove(self.inventory[0])
+        friend.remove(friend.inventory[0])
+        
+        return True
+
+    """
+    
+    Gets the item with the best condition in a certain category, one argument: a string that represents a category
+    
+    """
+    def get_best_by_category(self, category):
+        #looks through the instance's `inventory` for the item with the highest `condition` and matching `category`, returns this item
+        highestCondition = 0
+        bestItem = None
+        for item in self.inventory:
+            if item.category == category:
+                if item.condition > highestCondition:
+                    highestCondition = item.condition
+                    bestItem = item
+        return bestItem
+
+#If there are no items in the `inventory` that match the category, it returns `None`
+#It returns a single item even if there are duplicates (two or more of the same item with the same condition)
+
+
+
+
+
+#three arguments
+    #`other`, which represents another `Vendor` instance to trade with
+    #`my_priority`, which represents a category that the `Vendor` wants to receive
+    #`their_priority`, which represents a category that `other` wants to receive
+#the best item in my inventory that matches `their_priority` category is swapped with the best item in `other`'s inventory that matches `my_priority`
+    #It returns `True`
+    #If the `Vendor` has no item that matches `their_priority` category, swapping does not happen, and it returns `False`
+    #If `other` has no item that matches `my_priority` category, swapping does not happen, and it returns `False`
+
+    #swap the best item of certain categories with another `Vendor`
+    #oh my gosh I made my code so longgggg, could have used get best by category function and then the swap items function.....
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        myHighestCondition = 0
+        theirHighestCondition = 0
+        myBestItem = None
+        theirBestItem = None
+        
+        for myItem in self.inventory:
+            if myItem.category == their_priority:
+                if myItem.condition > myHighestCondition:
+                    myHighestCondition = myItem.condition
+                    myBestItem = myItem
+        
+        for theirItem in other.inventory:
+            if theirItem.category == my_priority:
+                if theirItem.condition > theirHighestCondition:
+                    theirHighestCondition = theirItem.condition
+                    theirBestItem = theirItem
+        
+        if myBestItem is None or theirBestItem is None:
+            return False
```

Comment:
I'm glad I saw this comment because I was about to provide the same feedback on utilizing `get_best_by_category` 😆